src/Tizen.NUI/src/internal/Common/WeakEvent.cs

```diff
@@ -21,9 +21,11 @@
 
 namespace Tizen.NUI
 {
-    internal class WeakEvent<T>
+    public class WeakEvent<T>
     {
         private List<WeakHandler<T>> handlers = new List<WeakHandler<T>>();
+        private bool isInInvoke = false;
+        private WeakHandler<T> pendingRemoval;
```

Comment:
```suggestion
        private HashSet<WeakHandler<T> pendingAdditional;
        private HashSet<WeakHandler<T> pendingRemoval;
```

I think we should allow to remove multiple items.

And also, we also have to 'block' the Add cases.

Reply:
Great point:) I will fix it soon